### PR TITLE
Add force path join func

### PR DESCRIPTION
**Pull request by jkonecny12**
This will always join the input paths even when the path starts with `/` which doesn't work with the `os.path.join` function.

This will be used by 
https://github.com/rhinstaller/anaconda/pull/2419
https://github.com/rhinstaller/anaconda/pull/2421

### CODE REVIEW

File: pyanaconda/core/util.py [hunk 1]
```diff
@@ -1258,6 +1258,27 @@ def sysroot_path(path):
     return os.path.join(conf.target.system_root, path.lstrip(os.path.sep))
 
 
+def force_path_join(path, *paths):
```
[the remark this answers is not in the document]
Comment (M4rtinK) [reply pyanaconda/core/util.py:1261]: Yeah `join_paths` (or `join_fs_paths`, etc.) sounds better to me as well.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Renamed to ``join_paths``
* Add code convention note.

**Comment by jkonecny12** [comment]
**Updated:**
* Fixed failing test after test module renaming. Thanks a lot @poncovka for your help!

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
**UPDATED:**
* Rebased to fix tests.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -131,6 +131,7 @@ In general we are trying to be as close as possible to `PEP8 <https://www.python
 * Never do wildcard (``from foo import *``) imports with the exception when all Anaconda developers agree on that.
 * Use ``raise`` & ``return`` in the doc string. Do not use ``raises`` or ``returns``.
 * Methods that return a task should have the suffix ‘_with_task’ (for example discover_with_task and DiscoverWithTask).
+* Prefer to use ``pyanaconda.util.join_paths`` over ``os.path.join``. See documentation for more info.
```
Comment (VladimirSlavik) [inline CONTRIBUTING.rst:134]: Maybe clarify that the documentation is on our function?

Comment (jkonecny12) [reply CONTRIBUTING.rst:134]: Isn't that obvious from the `pyanaconda` path?

Comment (VladimirSlavik) [reply CONTRIBUTING.rst:134]: Probably yes.

**Comment by VladimirSlavik** [comment]
Thank you!

**Pull request merged**